packages/react-dom/src/client/ReactDOMHostConfig.js

```diff
@@ -109,6 +110,8 @@ export type Container =
   | (Document & {_reactRootContainer?: FiberRoot, ...})
   | (DocumentFragment & {_reactRootContainer?: FiberRoot, ...});
 export type Instance = Element;
+export type InsertionParentInstance = Instance | DocumentFragment;
+export type InsertionInstance = Instance | Node;
```

Comment:
Need to bikeshed these names a bit but

Instance: Elements
InsertionParentInstance: A thing you can appendChild to. usually an Element but for Persisent HostComponents, a DocumentFragment
InsertionInstance: a Node in the DOM that we use to place other Instances against. Needs to be a Node b/c our insertion edge is a Comment atm.

It's hard to encode the fact that for Placement fibers the stateNode is a DocumentFragment until placement where it is swapped for an Instance. The nice thing about this is we have no branching except on Placement and Delete. But it requires types to be generalized. There are other places where I cheat and typecast to Instance b/c I don't want to loosen types and I know the current implementation is using apis derived from Node rather than Element

Reply:
Simplified as

Instance = the thing React knows how to insert
InstanceParent = something that can hold instances
InstanceSibling = something that can be used to position instances by adjacency (insertBefore(parent, child, nextSibling)